.github/workflows/main.yaml

```diff
@@ -35,3 +35,13 @@ jobs:
     with:
       charm-file-name: "sdcore-nrf_ubuntu-22.04-amd64.charm"
     secrets: inherit
+
+  publish-lib:
+    name: Publish Lib
+    needs:
+      [ publish-charm ]
+    if: ${{ github.ref_name == 'main' }}
```

Comment:
Should we also validate that the change was in the lib path? We probably don't want to call `charmcraft publish-lib` if no lib change was made

Reply:
The implementation seems a bit complicated, but it should do the trick.